### PR TITLE
New cluster page

### LINKED ISSUE #797: [1pt] Dashboard effector refactoring
Refactor cluster dashboard to effector and add unit tests to it

### PR DESCRIPTION
**Pull request by usenko-timur**
* Use effector for business logic and storing Cluster page data (models folder).
* Rewrite all Cluster page components using typescript.
* Add unit tests for the Failover modal.

- [x] Tests
- [x] Changelog
- [x] Documentation

Close #797

### CODE REVIEW

**Review by OlgaYakovleva — APPROVED** [review]
All tesst have been passed successfully, manually checks were successfully too

**Comment by aShelist** [comment]
webui/scripts/build.js → webui/scripts/_build.js
webui/scripts/start.js → webui/scripts/_start.js

Do we really need these files? Or can we delete?

**Comment by usenko-timur** [comment]
> webui/scripts/build.js → webui/scripts/_build.js webui/scripts/start.js → webui/scripts/_start.js
> 
> Do we really need these files? Or can we delete?

Deleted

File: webui/src/models/cluster/server-expel/index.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { $serverList, selectors } from '../server-list';
+
+// events
+export const serverExpelModalOpenEvent = app.domain.createEvent<{ uri: string }>('expel server modal open event');
+export const serverExpelModalCloseEvent = app.domain.createEvent('expel server modal close event');
+export const serverExpelEvent = app.domain.createEvent('expel server event');
+
+// stores
+export const $selectedServerExpelModalUri = app.domain.createStore<string | null>(null);
+export const $selectedServerExpelModalServer = sample({
```
Comment (aShelist) [inline webui/src/models/cluster/server-expel/index.ts:14]: I think, "combine" more suitable

File: webui/src/models/cluster/suggestions/index.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+export const disableServersApplyClick = app.domain.createEvent<unknown>('disable servers apply click');
+export const disableServersDetailsClick = app.domain.createEvent<unknown>('disable servers details click');
+export const restartReplicationsApplyClick = app.domain.createEvent<unknown>('restart replications apply click');
+export const restartReplicationsDetailsClick = app.domain.createEvent<unknown>('restart replications details click');
+export const forceApplyConfApplyClick = app.domain.createEvent<unknown>('force apply config apply click');
+export const forceApplyConfDetailsClick = app.domain.createEvent<unknown>('force apply config details click');
+export const forceApplyInstanceCheck = app.domain.createEvent<string>('force apply check instance');
+export const forceApplyReasonCheck = app.domain.createEvent<string>('force apply check instances with same reason');
+export const forceApplyReasonUncheck = app.domain.createEvent<string>('force apply uncheck instances with same reason');
+export const detailsClose = app.domain.createEvent('details modal close');
```
Comment (aShelist) [inline webui/src/models/cluster/suggestions/index.ts:26]: usually events were named with a postfix "Event"

File: webui/src/models/cluster/zones/index.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+// events
+export const setServerZoneEvent = app.domain.createEvent<{ uuid: string; zone?: string }>('set server sone click');
+export const zoneAddModalOpenEvent = app.domain.createEvent<{ uuid: string }>('zone adding modal open');
+export const zoneAddModalCloseEvent = app.domain.createEvent('zone adding modal close');
+export const zoneAddModalSetValueEvent = app.domain.createEvent<string>('zone adding modal set value');
+export const zoneAddModalSubmitEvent = app.domain.createEvent('add server zone modal submit');
+
+// stores
+export const $zoneAddModalError = app.domain.createStore<string>('');
+export const $zoneAddModalValue = restore(zoneAddModalSetValueEvent, '');
```
Comment (aShelist) [inline webui/src/models/cluster/zones/index.ts:16]: Is it specially created without a domain?

File: webui/src/pages/Cluster/Cluster.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import { ClusterControllers } from './ClusterControllers';
+import { ClusterModals } from './ClusterModals';
+import { ClusterPanels } from './ClusterPanels';
+import ButtonsPanel from './components/ButtonsPanel';
+import ReplicasetListPageSection from './components/ReplicasetListPageSection';
+import UnconfiguredServerListPageSection from './components/UnconfiguredServerListPageSection';
+
+const { $clusterPage } = cluster.page;
+
+const CLUSTER_PAGE_TITLE = 'Cluster';
```
Comment (aShelist) [inline webui/src/pages/Cluster/Cluster.tsx:20]: Maybe need created some exported constant? https://github.com/tarantool/cartridge/pull/1578/files#diff-a5d15763672cabe3ab885070ef25e3e3b8adef632e6a928d84025f2433c12796R13

**Review by aShelist — APPROVED** [review]

File: webui/src/store/effector/cluster/events.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { createEvent } from 'effector';
+
+import type { ClusterState } from './types';
+
+export const clusterPageMount = createEvent('cluster page mount');
```
[the remark this answers is not in the document]
Comment (usenko-timur) [reply webui/src/store/effector/cluster/events.ts:5]: Yes, but it's an old place.

File: webui/src/pages/ClusterLegacy/index.js [hunk 6]
```diff
@@ -130,8 +130,8 @@ class Cluster extends React.Component<ClusterProps> {
     const unlinkedServers = this.getUnlinkedServers();
 
     return (
-      <PageLayout heading="Cluster" headingContent={<ClusterButtonsPanel />}>
-        <AppTitle title="Cluster" />
+      <PageLayout heading="Cluster Legacy" headingContent={<ClusterButtonsPanel />}>
```
[the remark this answers is not in the document]
Comment (usenko-timur) [reply webui/src/pages/ClusterLegacy/index.js:133]: I left these files to test the new functionality and compare it with the old one.

File: webui/src/models/cluster/suggestions/index.ts (hunk 2 above)
Comment (usenko-timur) [reply webui/src/models/cluster/suggestions/index.ts:26]: This is the old functionality. I left it as is.

File: webui/src/api/graphql.js [hunk 7]
```diff
@@ -5,6 +5,10 @@ import { HttpLink } from 'apollo-link-http';
 
 import { getApiEndpoint } from 'src/apiEndpoints';
 
+import { getGraphqlError, getGraphqlErrorMessage, isGraphqlAccessDeniedError, isGraphqlErrorResponse } from './utils';
+
+export { getGraphqlError, getGraphqlErrorMessage, isGraphqlAccessDeniedError, isGraphqlErrorResponse };
```
[the remark this answers is not in the document]
Comment (usenko-timur) [reply webui/src/api/graphql.js:10]: Yes, it may need to be done in the future.

File: webui/src/models/cluster/zones/index.ts (hunk 3 above)
Comment (usenko-timur) [reply webui/src/models/cluster/zones/index.ts]: Thanks, fixed.

File: webui/src/pages/Cluster/Cluster.tsx (hunk 4 above)
Comment (usenko-timur) [reply webui/src/pages/Cluster/Cluster.tsx:20]: Fixed it a little differently.

File: webui/src/models/cluster/server-expel/index.ts (hunk 1 above)
Comment (usenko-timur) [reply webui/src/models/cluster/server-expel/index.ts]: Thanks, fixed.

**Pull request merged**